sematic/runners/tests/test_local_runner.py

```diff
@@ -53,6 +57,18 @@ def pipeline(a: float, b: float) -> float:
     return add(c, d)
 
 
+@func
+def do_cancel(x: float) -> float:
+    raise CancellationError("Fake cancellation")
+
+
+@func
+def cancelling_pipeline(a: float, b: float) -> float:
+    c = add(a, b)
+    d = do_cancel(add3(a, b, c))
+    return add(c, d)
```

Comment:
Please replace this `add` with a func that raises an exception (which would fail the test), so that the test also validates the funcs downstream from the cancelation func are not actually executed.